code/modules/projectiles/guns/projectile.dm

```diff
@@ -212,10 +212,10 @@
 //attempts to unload src. If allow_dump is set to 0, the speedloader unloading method will be disabled
 /obj/item/weapon/gun/projectile/proc/unload_ammo(mob/user, var/allow_dump=1)
 	if(ammo_magazine)
-		if(!user.put_in_hands(ammo_magazine))
```

Comment:
This line meant that if put_in_hands returned false, the rest of the proc wouldn't run. If put_in_hands put it on the tile beneath the user (because their hands are full) put_in_hands would return false, meaning the magazine would drop to the ground but not get unliked through the rest of the proc below.